Week02/lesson04/homework/relaxr_starter_code/css/style.css

```diff
@@ -0,0 +1,200 @@
+/*
+Project Name: Relaxr Homepage
+Client: Relaxr
+Author: Christophe Brumby
+Developer @ GA in London
+*/
+
+
+/******************************************
+/* SETUP
+/*******************************************/
+
+/* Box Model Hack */
+* {
+     -moz-box-sizing: border-box; /* Firexfox */
+     -webkit-box-sizing: border-box; /* Safari/Chrome/iOS/Android */
+     box-sizing: border-box; /* IE */
+}
+
+/* Clear fix hack */
+.clearfix:after {
+     content: ".";
+     display: block;
+     clear: both;
+     visibility: hidden;
+     line-height: 0;
+     height: 0;
+}
+
+.clear {
+  clear: both;
+}
+
+.alignright {
+  float: right;
+  padding: 0 0 1% 1%; /* note the padding around a right floated image */
+  width: 50%;
+  text-align: left;
+}
+
+.alignleft {
+  float: left;
+  padding: 0 1% 1% 0; /* note the padding around a left floated image */
+  width: 50%;
+  text-align: right;
+}
+
+/******************************************
+/* BASE STYLES
+/*******************************************/
+
+body {
+     color: #000;
+     font-size: 12px;
+     line-height: 1.4;
+     font-family: Open Sans, Helvetica, Arial, sans-serif;
+     background: url(../images/header_bg.jpg) no-repeat center top rgb(134,141,157);
+      background-size: contain;
+}
+
+
+/******************************************
+/* LAYOUT
+/*******************************************/
+
+/* Center the container */
+#container {
+  width: auto;
+}
+
+header {
+  padding-bottom: 22%;
+}
+
+footer {
+  padding: 3% 0 1%;
+  background-color: #121212;
+}
+
+/******************************************
+/* ADDITIONAL STYLES
+/*******************************************/
+
+h1 {
+  font-weight: 800;
+  font-size: 24px;
+  color: #FFF;
+  margin: 1% 0 8% 6%;
```

Comment:
Ok so it's a marginal thing, but your code is __really__ good so I'll get into some advanced CSS concepts if that's ok with you :)

Right, so there's the concept of "bleeding margins" from within children elements. We very briefly saw this in class when you were getting that weird margin at the top of your page that was coming from way down the DOM tree, from a grandchild of `<body>`, if I recall correctly. Basically, the idea is, if you need separation between two things on the page, then you put the margins at that level, and not further down the DOM tree.

Let's look at this part of your code. You've got some HTML that basically looks like this:

```
<header>
  <h1>Relaxr</h1>
  <h2>Get piece of mind with a single tap</h2>
  <a>...Get it Now...</a>
</header>
```

If I highlight the `<header>` in the Chrome inspector you'll notice a gap between the edge of the `<header>` and the top of the screen:

![image](https://cloud.githubusercontent.com/assets/5670279/17462063/632cea9e-5c98-11e6-9f1b-bd536004419a.png)

...this is due to margin bleed from the child within `<header>`, the `<h1>`.

So, first and foremost we need stop that from happening. As a general rule, if you want to create some separation between a child and its parent, it's better to use padding, so I'm first going to kill all the `margin` you've put on the `<h1>` and make the `top` and `left` (which are the sides that need separation from the parent) use `padding` instead using `padding: 1% 0 0 6%;`

So now, if I select the `<header>` in the Chrome inspector, the result is like this:

![image](https://cloud.githubusercontent.com/assets/5670279/17462094/5611d27e-5c99-11e6-84be-0d5c26f37725.png)

...cool! No bleeding margins. The child (the `<h1>`) looks like this:

![image](https://cloud.githubusercontent.com/assets/5670279/17462097/652fe0a2-5c99-11e6-8ef7-5049239d6225.png)

...also good, the padding creates a nice separation from the top and left.

But now you want add some space between the `<h1>` and the `<h2>Get piece of mind with a single tap</h2>`. At this point, you __do__ want to use `margin` (though really you could use padding too) because you want to *create separation between two elements on the same hierarchy level.*

So let's summarise:

__Separate child from edges of parent__ = use `padding`
__Separate two elements on the same hierarchy level of the DOM tree__ = use `margin`

.. ok so back to your code, we want to separate `<h1>` and `<h2>` and they both sit at the same level of the DOM tree, so let's use margin. Your `<h1>` css should now look like this:

```
h1 {
  padding: 1% 0 0 6%;
  margin-bottom: 8%;
  ...etc..
}
```

Now if I select the `<h1>` in the Chrome dev tools, we're back in business:

![image](https://cloud.githubusercontent.com/assets/5670279/17462112/73d4b6e0-5c9a-11e6-8024-c40469839b11.png)

Like I said this is a minor thing, and it's a pretty advanced CSS concept, but it'll help you with more complex layouts in future. Knowing when to use `padding` and when to use `margin` is something I still have to think about, it's not always really obvious. But you definitely want to avoid children bleeding margins, because especially around the time we start doing Responsive CSS, it'll really come back to bite you in the bum.

Sorry for the long explanation, please tell me if something wasn't clear.